packages/thirdweb/src/wallets/ecosystem/get-ecosystem-wallet-info.ts

```diff
@@ -0,0 +1,55 @@
+import { getThirdwebBaseUrl } from "../../utils/domains.js";
+import type { Prettify } from "../../utils/type-utils.js";
+import type { WalletInfo } from "../wallet-info.js";
+import type { EcosystemWalletId } from "../wallet-types.js";
+
+/**
+ * Fetches metadata for a given ecosystem wallet.
+ *
+ * @param {EcosystemWalletId} walletId - The ecosystem wallet ID.
+ * @returns {Promise<{ imageUrl: string | undefined, name: string | undefined }>} A promise that resolves to an object containing the wallet's image URL and name.
+ * @throws {Error} Throws an error if no partner ID is provided in the wallet configuration.
+ * @internal
+ */
+export async function getEcosystemWalletInfo(
+  walletId: EcosystemWalletId,
+): Promise<Prettify<WalletInfo>> {
+  const headers = new Headers();
+  headers.set("x-ecosystem-partner-id", "80ea22d8-7471-43e9-9f20-961f6fa88771");
+  const res = await fetch(
```

Comment:
should we use our clientFetch here to get the sdk/platform headers too?

Reply:
Now realizing we'd have to also pass the client to getWalletInfo upstream since it calls this, is that worth it?

Reply:
Is the API not gated? Like it's fully public?